shell-ui/src/index.js

```diff
@@ -0,0 +1,194 @@
+//@flow
+import React, { useEffect, useLayoutEffect, useRef } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import reactToWebComponent from 'react-to-webcomponent';
+import { ThemeProvider as StyledComponentsProvider } from 'styled-components';
+import { WebStorageStateStore } from 'oidc-client';
+import { AuthProvider, AuthProviderProps, UserManager } from 'oidc-react';
+import { QueryClient, QueryClientProvider, useQuery } from 'react-query';
+import { LoadingNavbar, Navbar } from './NavBar';
+import { UserDataListener } from './UserDataListener';
+import { logOut } from './auth/logout';
+
+const EVENTS_PREFIX = 'solutions-navbar--';
+export const AUTHENTICATED_EVENT: string = EVENTS_PREFIX + 'authenticated';
+
+type Options = { [path: string]: { en: string, fr: string, roles?: string[] } }; // TODO should be able to accept configs for paths in dropdown menu under user name
+
+export type SolutionsNavbarProps = {
+  'oidc-provider-url'?: string,
+  scopes?: string,
+  'client-id'?: string,
+  'response-type'?: string,
+  'redirect-url'?: string,
+  options?: Options,
+  onAuthenticated?: (evt: CustomEvent) => void,
+  logOut?: () => void,
+  setUserManager?: (userManager: UserManager) => void,
+};
+
+type Config = {
+  docUrl?: string,
+  oidc?: {
+    providerUrl?: string,
+    redirectUrl?: string,
+    clientId?: string,
+    responseType?: string,
+    scopes?: string,
+  },
+  options?: Options,
+};
+
+const SolutionsNavbar = ({
+  'oidc-provider-url': oidcProviderUrl,
+  scopes,
+  'client-id': clientId,
+  'redirect-url': redirectUrl,
+  'response-type': responseType,
+  options,
+  onAuthenticated,
+  logOut,
+  setUserManager,
+}: SolutionsNavbarProps) => {
+  const { data: config, status } = useQuery<Config>('navbarConfig', () =>
+    fetch('/shell/config.json').then((r) => {
```

Comment:
This path depends on where the navbar is deployed / included, I think, so likely something that needs to be configurable.
Also, since all configuration options can be overridden by props, would it make sense to make this config file optional?